static/js/compose.js

```diff
@@ -299,6 +308,7 @@ export function finish() {
     if (!compose_validate.validate()) {
         // If the message failed validation, hide compose spinner.
         compose_ui.hide_compose_spinner();
```

Comment:
Hmm, I would have thought that the compose spinner experience would have preventing sending again. So it's possible we don't need a new state variable, but to fix a bug with the existing one.

(I seem to recall we have a similar spinner experience for message edit?)

Reply:
> (I seem to recall we have a similar spinner experience for message edit?)

Yes, but a double send bug there is not possible.

Reply:
@sahil839 can I get your thoughts on the approach here? I remember you worked on the spinner experience for message edit, and I'm wondering if we'd be better off using a design like what we did there for this particular detail.

Reply:
We are not disabling the button while showing the spinner and in message-edit case we set `pointer-events: none` so I guess click is just ignored in that case. Can we just directly disable the button while message is being sent in case of double clicks?

Reply:
@amanagr thoughts on that proposed alternative approach?

I guess one thing to keep in mind is keyboard UI would not be affected by the button state, but I'm not sure double-send via the keyboard UI was possible anyway?

Reply:
You could press the send button and press enter to reproduce the bug.

Reply:
OK, I'm fine with doing a state variable here.